internal/graph/resolvers/composite.go

```diff
@@ -215,7 +215,9 @@ func (r *compositeResourceSpec) Resources(ctx context.Context, obj *model.Compos
 			xrc.SetKind(ref.Kind)
 			nn := types.NamespacedName{Namespace: ref.Namespace, Name: ref.Name}
 			if err := c.Get(ctx, nn, xrc); err != nil {
-				graphql.AddError(ctx, errors.Wrap(err, errGetComposed))
+				if !apierrors.IsNotFound(err) {
```

Comment:
An alternative would be to not consider not found a GraphQL error. We could either return a union that is the found object or not found maybe including the object ref or we could just not include not found refs as you have it here and rely on the consumer to compare with the objectrefs field to see what wasn't fetched.

Reply:
for context, other refs are handled exactly this way:
* https://github.com/upbound/xgql/blob/main/internal/graph/resolvers/composite.go#L127-L129
* https://github.com/upbound/xgql/blob/main/internal/graph/resolvers/composite.go#L168C3-L170
* https://github.com/upbound/xgql/blob/main/internal/graph/resolvers/composite.go#L127-L129